coremltools/converters/mil/mil/ops/defs/iOS15/linear.py

```diff
@@ -289,7 +289,8 @@ def type_inference(self):
         x_shape = x.shape
         y_shape = y.shape
         assert len(x_shape) == len(y_shape), "inputs not of the same rank"
-        assert x_shape[-1] == y_shape[-3], "input shapes incompatible"
+        assert x_shape[-1] == y_shape[-3] or is_symbolic(x_shape[-1]) or is_symbolic(y_shape[-3]), \
```

Comment:
nit: Would it be more readable if we change it to the following?
```
if not (is_symbolic(x_shape[-1]) or is_symbolic(y_shape[-3])):
    assert x_shape[-1] == y_shape[-3], f"input shapes incompatible: {x_shape[-1]} and {y_shape[-3]}"
```

Reply:
Done